<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>2250</width>
    <height>1280</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>181</width>
      <height>281</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-image: url(:/pic/logotruong.png);</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>70</y>
      <width>1411</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>28</pointsize>
     </font>
    </property>
    <property name="text">
     <string>HỆ THỐNG PHÂN LOẠI LINH KIỆN ĐIỆN TỬ</string>
    </property>
   </widget>
   <widget class="QComboBox" name="serialPort">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>1050</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QComboBox {
    background-color: #D11D2F; /* Nền đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 2px solid #CD5C5C; /* Viền đỏ đậm */
    padding: 5px;              /* Khoảng cách nội dung */
    padding-right: 30px;       /* Tạo không gian cho mũi tên */
}

QComboBox:hover {
    background-color: #FF6347; /* Màu đỏ sáng hơn khi rê chuột */
}

QComboBox:focus {
    border: 2px solid #8B0000; /* Viền đậm hơn khi được chọn */
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right; /* Vị trí nút mũi tên */
    border: none;                   /* Bỏ viền nút mũi tên */
    background-color: #B22222;      /* Màu nền mũi tên */
    border-top-right-radius: 20px;  /* Bo tròn góc trên phải */
    border-bottom-right-radius: 20px; /* Bo tròn góc dưới phải */
    width: 30px;                    /* Độ rộng của nút mũi tên */
}

QComboBox::down-arrow {
    width: 0;
    height: 0;
    border-left: 6px solid transparent; /* Tạo cạnh trái của tam giác */
    border-right: 6px solid transparent; /* Tạo cạnh phải của tam giác */
    border-top: 6px solid white;        /* Tạo cạnh trên màu trắng */
    margin: 5px;                        /* Căn chỉnh mũi tên trong nút */
}

QComboBox QAbstractItemView {
    background-color: #FFF5F5; /* Nền danh sách thả xuống */
    border: 1px solid #CD5C5C; /* Viền đỏ nhạt */
    selection-background-color: #B22222; /* Nền khi chọn */
    selection-color: white;             /* Chữ khi chọn */
}
</string>
    </property>
   </widget>
   <widget class="QComboBox" name="modeSelect">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>1150</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QComboBox {
    background-color: #D11D2F; /* Nền đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 2px solid #CD5C5C; /* Viền đỏ đậm */
    padding: 5px;              /* Khoảng cách nội dung */
    padding-right: 30px;       /* Tạo không gian cho mũi tên */
}

QComboBox:hover {
    background-color: #FF6347; /* Màu đỏ sáng hơn khi rê chuột */
}

QComboBox:focus {
    border: 2px solid #8B0000; /* Viền đậm hơn khi được chọn */
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right; /* Vị trí nút mũi tên */
    border: none;                   /* Bỏ viền nút mũi tên */
    background-color: #B22222;      /* Màu nền mũi tên */
    border-top-right-radius: 20px;  /* Bo tròn góc trên phải */
    border-bottom-right-radius: 20px; /* Bo tròn góc dưới phải */
    width: 30px;                    /* Độ rộng của nút mũi tên */
}

QComboBox::down-arrow {
    width: 0;
    height: 0;
    border-left: 6px solid transparent; /* Tạo cạnh trái của tam giác */
    border-right: 6px solid transparent; /* Tạo cạnh phải của tam giác */
    border-top: 6px solid white;        /* Tạo cạnh trên màu trắng */
    margin: 5px;                        /* Căn chỉnh mũi tên trong nút */
}

QComboBox QAbstractItemView {
    background-color: #FFF5F5; /* Nền danh sách thả xuống */
    border: 1px solid #CD5C5C; /* Viền đỏ nhạt */
    selection-background-color: #B22222; /* Nền khi chọn */
    selection-color: white;             /* Chữ khi chọn */
}
</string>
    </property>
   </widget>
   <widget class="QPushButton" name="startSystem">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>1050</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>START</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopSystem">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>1150</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>STOP</string>
    </property>
   </widget>
   <widget class="QPushButton" name="startBtn_2">
    <property name="geometry">
     <rect>
      <x>1880</x>
      <y>1100</y>
      <width>201</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>HISTORY</string>
    </property>
   </widget>
   <widget class="QPushButton" name="openCamera">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>1050</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>OPEN</string>
    </property>
   </widget>
   <widget class="QPushButton" name="closeCamera">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>1150</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>CLOSE</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableComponents">
    <property name="geometry">
     <rect>
      <x>1250</x>
      <y>330</y>
      <width>811</width>
      <height>671</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QTableWidget {
    border: 2px solid #D11D2F; /* Viền màu đỏ */
    gridline-color: #D11D2F;  /* Màu đường lưới bên trong */
    background-color: #FFFFFF; /* Màu nền bảng */
    selection-background-color: #F5B7B1; /* Màu nền khi chọn ô */
    selection-color: black; /* Màu chữ khi chọn ô */
}

QHeaderView::section {
    background-color: #D11D2F; /* Màu nền tiêu đề cột/hàng */
    color: white;             /* Màu chữ của tiêu đề */
    border: 1px solid #FFFFFF; /* Đường viền giữa các tiêu đề */
}

QTableCornerButton::section {
    background-color: #D11D2F; /* Màu nền góc trên bên trái */
    border: 2px solid #D11D2F; /* Đường viền của góc */
}
</string>
    </property>
    <property name="lineWidth">
     <number>10</number>
    </property>
    <property name="iconSize">
     <size>
      <width>100</width>
      <height>100</height>
     </size>
    </property>
    <property name="columnCount">
     <number>2</number>
    </property>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>100</number>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>300</number>
    </attribute>
    <attribute name="verticalHeaderMinimumSectionSize">
     <number>10</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>300</number>
    </attribute>
    <row>
     <property name="text">
      <string>IMAGE</string>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </row>
    <row>
     <property name="text">
      <string>QUANTITY</string>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </row>
    <column>
     <property name="text">
      <string>Điện trở</string>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Tụ điện</string>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </column>
    <item row="0" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="videoLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>981</width>
      <height>671</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopSystem_2">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>1050</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>AUTO</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopSystem_3">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>1150</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>MANUAL</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopSystem_4">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>1050</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>SERVO1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopSystem_5">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>1150</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #D11D2F; /* Màu đỏ sáng khi không nhấn */
    color: white;              /* Màu chữ */
    border-radius: 20px;       /* Bo tròn góc */
    border: 4px solid black;   /* Viền đen */
}

QPushButton:pressed {
    background-color: #B22222; /* Màu đỏ đậm khi nhấn */
    border: 4px solid black;   /* Viền đen khi nhấn */
}
</string>
    </property>
    <property name="text">
     <string>SERVO2</string>
    </property>
   </widget>
   <widget class="QDial" name="dial">
    <property name="geometry">
     <rect>
      <x>1490</x>
      <y>1060</y>
      <width>171</width>
      <height>151</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1650</x>
      <y>1110</y>
      <width>101</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>128</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>2250</width>
     <height>33</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="logoTruong.qrc"/>
 </resources>
 <connections/>
</ui>
